### ISSUE TITLE
4.0.2 bash install on Raspberry Pi 4 Model B "failed to build cryptography"

### ISSUE THREAD

**Issue by SimUKdev** [thonny/thonny#2634]
I have been using Thonny 4.0.0 and 4.0.1 on a Raspberry Pi 4 Model B for a while, installed via the  `bash <(wget -O - https://thonny.org/installer-for-linux)` method as it's ahead of the Raspberry Pi OS version.

I wanted to update to 4.0.2 so repeated the same install procedure over the top of an existing install, and it failed with the "failed to build cryptography" and a lot of red text. To rule the "upgrading an existing install"  out of the reason for being the cause, and also to rule out the OS itself, I imaged an all new install of Raspberry Pi OS 11 Bullseye using the Raspberry Pi official imager.

On that new installation I did the usual `sudo apt get update` and `sudo apt get upgrade` to ensure it was fully up to date. I then did `sudo apt remove thonny` to remove the built-in version (which I believe is/was 4.0.1), after it was removed I did the `bash` install on that freshly imaged and updated Pi, but still got the same error: *(sorry for the long copy/paste of the terminal output)*

```
$ bash <(wget -O - https://thonny.org/installer-for-linux)

Redirecting output to ‘wget-log’.
Thonny only provides pre-built bundles for x86_64 (not armv7l).
I could install Thonny into a venv under your existing Python 3.9.2 (/usr/bin/python3).
Press ENTER to continue or Ctrl+C to cancel.
Creating virtual environment for Thonny
Marking virtual environment as private to Thonny
Installing Thonny's dependencies
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Ignoring adafruit-board-toolkit: markers 'sys_platform == "win32" or sys_platform == "darwin"' don't match your environment
Collecting jedi==0.18.*
  Downloading https://www.piwheels.org/simple/jedi/jedi-0.18.2-py2.py3-none-any.whl (1.6 MB)
     |████████████████████████████████| 1.6 MB 1.7 MB/s 
Collecting pyserial==3.5
  Downloading https://www.piwheels.org/simple/pyserial/pyserial-3.5-py2.py3-none-any.whl (90 kB)
     |████████████████████████████████| 90 kB 1.6 MB/s 
Collecting pylint==2.15.*
  Downloading https://www.piwheels.org/simple/pylint/pylint-2.15.10-py3-none-any.whl (509 kB)
     |████████████████████████████████| 509 kB 2.7 MB/s 
Collecting docutils==0.19.*
  Downloading https://www.piwheels.org/simple/docutils/docutils-0.19-py3-none-any.whl (570 kB)
     |████████████████████████████████| 570 kB 3.3 MB/s 
Collecting mypy==0.991.*
  Downloading https://www.piwheels.org/simple/mypy/mypy-0.991-py3-none-any.whl (2.3 MB)
     |████████████████████████████████| 2.3 MB 3.1 MB/s 
Collecting asttokens==2.2.*
  Downloading https://www.piwheels.org/simple/asttokens/asttokens-2.2.1-py2.py3-none-any.whl (26 kB)
Collecting Send2Trash==1.8.1b0
  Downloading https://www.piwheels.org/simple/send2trash/Send2Trash-1.8.1b0-py3-none-any.whl (17 kB)
Collecting esptool==4.4.*
  Downloading https://www.piwheels.org/simple/esptool/esptool-4.4-py3-none-any.whl (288 kB)
     |████████████████████████████████| 288 kB 4.2 MB/s 
Collecting bcrypt==3.2.*
  Downloading https://www.piwheels.org/simple/bcrypt/bcrypt-3.2.2-cp39-cp39-linux_armv7l.whl (51 kB)
     |████████████████████████████████| 51 kB 154 kB/s 
Collecting cryptography==38.*
  Downloading cryptography-38.0.4.tar.gz (599 kB)
     |████████████████████████████████| 599 kB 1.0 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting paramiko==2.12.*
  Downloading https://www.piwheels.org/simple/paramiko/paramiko-2.12.0-py2.py3-none-any.whl (213 kB)
     |████████████████████████████████| 213 kB 428 kB/s 
Collecting websockets==10.*
  Downloading https://www.piwheels.org/simple/websockets/websockets-10.4-cp39-cp39-linux_armv7l.whl (105 kB)
     |████████████████████████████████| 105 kB 1.1 MB/s 
Collecting ptyprocess==0.7.*
  Downloading https://www.piwheels.org/simple/ptyprocess/ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Collecting six
  Downloading https://www.piwheels.org/simple/six/six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting cffi>=1.1
  Using cached https://www.piwheels.org/simple/cffi/cffi-1.15.1-cp39-cp39-linux_armv7l.whl (364 kB)
Collecting ecdsa>=0.16.0
  Downloading https://www.piwheels.org/simple/ecdsa/ecdsa-0.18.0-py2.py3-none-any.whl (142 kB)
     |████████████████████████████████| 142 kB 399 kB/s 
Collecting reedsolo<=1.5.4,>=1.5.3
  Downloading https://www.piwheels.org/simple/reedsolo/reedsolo-1.5.4-cp39-cp39-linux_armv7l.whl (921 kB)
     |████████████████████████████████| 921 kB 1.2 MB/s 
Collecting bitstring<4,>=3.1.6
  Downloading https://www.piwheels.org/simple/bitstring/bitstring-3.1.9-py3-none-any.whl (39 kB)
Collecting parso<0.9.0,>=0.8.0
  Downloading https://www.piwheels.org/simple/parso/parso-0.8.3-py2.py3-none-any.whl (100 kB)
     |████████████████████████████████| 100 kB 1.4 MB/s 
Collecting mypy-extensions>=0.4.3
  Downloading https://www.piwheels.org/simple/mypy-extensions/mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Collecting typing-extensions>=3.10
  Using cached https://www.piwheels.org/simple/typing-extensions/typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting tomli>=1.1.0
  Downloading https://www.piwheels.org/simple/tomli/tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting pynacl>=1.0.1
  Downloading https://www.piwheels.org/simple/pynacl/PyNaCl-1.5.0-cp39-cp39-linux_armv7l.whl (585 kB)
     |████████████████████████████████| 585 kB 591 kB/s 
Collecting tomlkit>=0.10.1
  Downloading https://www.piwheels.org/simple/tomlkit/tomlkit-0.11.6-py3-none-any.whl (35 kB)
Collecting mccabe<0.8,>=0.6
  Downloading https://www.piwheels.org/simple/mccabe/mccabe-0.7.0-py2.py3-none-any.whl (7.3 kB)
Collecting platformdirs>=2.2.0
  Downloading https://www.piwheels.org/simple/platformdirs/platformdirs-2.6.2-py3-none-any.whl (14 kB)
Collecting isort<6,>=4.2.5
  Downloading https://www.piwheels.org/simple/isort/isort-5.12.0-py3-none-any.whl (91 kB)
     |████████████████████████████████| 91 kB 837 kB/s 
Collecting astroid<=2.14.0-dev0,>=2.12.13
  Downloading https://www.piwheels.org/simple/astroid/astroid-2.13.5-py3-none-any.whl (273 kB)
     |████████████████████████████████| 273 kB 2.2 MB/s 
Collecting dill>=0.2
  Downloading https://www.piwheels.org/simple/dill/dill-0.3.6-py3-none-any.whl (110 kB)
     |████████████████████████████████| 110 kB 3.8 MB/s 
Collecting wrapt<2,>=1.11
  Downloading https://www.piwheels.org/simple/wrapt/wrapt-1.14.1-cp39-cp39-linux_armv7l.whl (72 kB)
     |████████████████████████████████| 72 kB 397 kB/s 
Collecting lazy-object-proxy>=1.4.0
  Downloading https://www.piwheels.org/simple/lazy-object-proxy/lazy_object_proxy-1.9.0-cp39-cp39-linux_armv7l.whl (60 kB)
     |████████████████████████████████| 60 kB 208 kB/s 
Collecting pycparser
  Using cached https://www.piwheels.org/simple/pycparser/pycparser-2.21-py2.py3-none-any.whl (119 kB)
Building wheels for collected packages: cryptography
  Building wheel for cryptography (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/pi/apps/thonny/bin/python3 /tmp/tmpvfsi1uhv_in_process.py build_wheel /tmp/tmpcr0urxtj
       cwd: /tmp/pip-install-39zrc8fo/cryptography_46a7e0be21e544eaa12307399518e475
  Complete output (193 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-armv7l-cpython-39
  creating build/lib.linux-armv7l-cpython-39/cryptography
  copying src/cryptography/__about__.py -> build/lib.linux-armv7l-cpython-39/cryptography
  copying src/cryptography/fernet.py -> build/lib.linux-armv7l-cpython-39/cryptography
  copying src/cryptography/exceptions.py -> build/lib.linux-armv7l-cpython-39/cryptography
  copying src/cryptography/utils.py -> build/lib.linux-armv7l-cpython-39/cryptography
  copying src/cryptography/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat
  copying src/cryptography/hazmat/_oid.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat
  copying src/cryptography/hazmat/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat
  creating build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/base.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/name.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/oid.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/certificate_transparency.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/general_name.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/ocsp.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/extensions.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  copying src/cryptography/x509/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/x509
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings
  copying src/cryptography/hazmat/bindings/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends
  copying src/cryptography/hazmat/backends/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/hmac.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/_asymmetric.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/constant_time.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/_serialization.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/cmac.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/hashes.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/padding.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/keywrap.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/poly1305.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/_cipheralgorithm.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/binding.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/_conditional.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/openssl
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x25519.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/dh.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x448.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/hmac.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/cmac.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ec.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/hashes.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/decode_asn1.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/utils.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/rsa.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/aead.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ed448.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/dsa.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x509.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/backend.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/poly1305.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ed25519.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ciphers.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/backends/openssl
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/pkcs12.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/base.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/pkcs7.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/ssh.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/serialization
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/scrypt.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/kbkdf.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/x963kdf.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/concatkdf.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/pbkdf2.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/hkdf.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/kdf
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/hotp.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/totp.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/twofactor
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/algorithms.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/base.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/aead.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/modes.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/ciphers
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/x25519.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/dh.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/x448.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/types.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ec.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/utils.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/rsa.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ed448.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/dsa.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/padding.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ed25519.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/__init__.py -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/primitives/asymmetric
  running egg_info
  writing src/cryptography.egg-info/PKG-INFO
  writing dependency_links to src/cryptography.egg-info/dependency_links.txt
  writing requirements to src/cryptography.egg-info/requires.txt
  writing top-level names to src/cryptography.egg-info/top_level.txt
  reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  no previously-included directories found matching 'docs/_build'
  warning: no previously-included files found matching 'vectors'
  warning: no previously-included files matching '*' found under directory 'vectors'
  warning: no previously-included files matching '*' found under directory '.github'
  warning: no previously-included files found matching 'release.py'
  warning: no previously-included files found matching '.readthedocs.yml'
  warning: no previously-included files found matching 'dev-requirements.txt'
  warning: no previously-included files found matching 'tox.ini'
  warning: no previously-included files found matching 'mypy.ini'
  warning: no previously-included files matching '*' found under directory '.circleci'
  adding license file 'LICENSE'
  adding license file 'LICENSE.APACHE'
  adding license file 'LICENSE.BSD'
  adding license file 'LICENSE.PSF'
  writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
  /tmp/pip-build-env-vgibdifx/overlay/lib/python3.9/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'cryptography.hazmat.bindings._rust' as data is deprecated, please list it in `packages`.
      !!
  
  
      ############################
      # Package would be ignored #
      ############################
      Python recognizes 'cryptography.hazmat.bindings._rust' as an importable package,
      but it is not listed in the `packages` configuration of setuptools.
  
      'cryptography.hazmat.bindings._rust' has been automatically added to the distribution only
      because it may contain data files, but this behavior is likely to change
      in future versions of setuptools (and therefore is considered deprecated).
  
      Please make sure that 'cryptography.hazmat.bindings._rust' is included as a package by using
      the `packages` configuration field or the proper discovery methods
      (for example by using `find_namespace_packages(...)`/`find_namespace:`
      instead of `find_packages(...)`/`find:`).
  
      You can read more about "package discovery" and "data files" on setuptools
      documentation page.
  
  
  !!
  
    check.warn(importable)
  copying src/cryptography/py.typed -> build/lib.linux-armv7l-cpython-39/cryptography
  copying src/cryptography/hazmat/bindings/_openssl.pyi -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings
  creating build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/_rust
  copying src/cryptography/hazmat/bindings/_rust/__init__.pyi -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/_rust
  copying src/cryptography/hazmat/bindings/_rust/asn1.pyi -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/_rust
  copying src/cryptography/hazmat/bindings/_rust/ocsp.pyi -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/_rust
  copying src/cryptography/hazmat/bindings/_rust/x509.pyi -> build/lib.linux-armv7l-cpython-39/cryptography/hazmat/bindings/_rust
  running build_ext
  generating cffi module 'build/temp.linux-armv7l-cpython-39/cryptography.hazmat.bindings._openssl.c'
  creating build/temp.linux-armv7l-cpython-39
  running build_rust
  
      =============================DEBUG ASSISTANCE=============================
      If you are seeing a compilation error please try the following steps to
      successfully install cryptography:
      1) Upgrade to the latest pip and try again. This will fix errors for most
         users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
      2) Read https://cryptography.io/en/latest/installation/ for specific
         instructions for your platform.
      3) Check our frequently asked questions for more information:
         https://cryptography.io/en/latest/faq/
      4) Ensure you have a recent Rust toolchain installed:
         https://cryptography.io/en/latest/installation/#rust
  
      Python: 3.9.2
      platform: Linux-5.15.61-v7l+-armv7l-with-glibc2.31
      pip: n/a
      setuptools: 67.1.0
      setuptools_rust: 1.5.2
      rustc: n/a
      =============================DEBUG ASSISTANCE=============================
  
  error: can't find Rust compiler
  
  If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
  
  To update pip, run:
  
      pip install --upgrade pip
  
  and then retry package installation.
  
  If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
  
  This package requires Rust >=1.48.0.
  ----------------------------------------
  ERROR: Failed building wheel for cryptography
Failed to build cryptography
ERROR: Could not build wheels for cryptography which use PEP 517 and cannot be installed directly
```

The text becomes red shortly after the (successful) pycparser stage, at the "ERROR:" line:

```
Building wheels for collected packages: cryptography
  Building wheel for cryptography (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
```

I have followed the advised `pip install --upgrade pip` and that updates pip ("*Successfully installed pip-23.0*"), but still get the same error when trying to install the Thonny 4.0.2 build as before.

```
(ERROR: Failed building wheel for cryptography
Failed to build cryptography
ERROR: Could not build wheels for cryptography which use PEP 517 and cannot be installed directly)
```

I've not tried doing anything with Rust as also mentioned in the error message, and didn't have to do anything specific with Rust on a previous earlier Thonny 4.x install via the Bash, but it does seem from the error messages to be relating to something to do with (the lack of) that.

**Comment by aivarannamaa** [comment]
Thank you for the report!

Thonny's installer scripts asks for `cryptography==38.*`, which means `pip` tries to install 38.0.4, but it seems that the build for this version has failed at [piwheels](https://www.piwheels.org/project/cryptography/).

Please try replacing `cryptography==38.*` with `cryptography==38.0.1` in the bash script and launching it again.

**Comment by aivarannamaa** [comment]
Works with just released 4.1.0
